includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -428,13 +458,23 @@ public function allowed_items_in_cart() {
 				return false;
 			}
 
+			// Not supported when user isn't authenticated and authentication is required.
+			if ( ! is_user_logged_in() && $this->is_authentication_required() ) {
```

Comment:
This block can be moved outside the `foreach` block since it applies to the whole cart.

Reply:
Well flagged. I ended up moving it to a more appropriate position, in `should_show_payment_button_on_cart`.